### PR TITLE
sqlliveness: wait for migration before starting loops

### PR DESCRIPTION
**Pull request by ajwerner**
Prior to this commit we'd hit some errors which would get retried
if the sqlmigration was slow. Better to just wait for it to happen.

Also sneaks in some changes to use the context rather than quiescence
channel because it makes the waiting cleaner.

Release note: None

### CODE REVIEW

File: pkg/sql/sqlliveness/sqlliveness.go [hunk 1]
```diff
@@ -78,3 +81,22 @@ type Reader interface {
 	// Instance that is attempting to claim expired resources.
 	IsAlive(context.Context, SessionID) (alive bool, err error)
 }
+
+// WaitForActive waits for the sqlliveness subsystem's migration to have been
+// performed.
+func WaitForActive(ctx context.Context, settings *cluster.Settings) {
+	for r := retry.StartWithCtx(ctx, retry.Options{}); r.Next(); {
```
Comment (spaskob) [inline pkg/sql/sqlliveness/sqlliveness.go:88]: why empty options?

File: pkg/sql/sqlliveness/slinstance/slinstance.go [hunk 2]
```diff
@@ -250,7 +250,7 @@ func (l *Instance) Start(ctx context.Context) {
 		return
 	}
 	log.Infof(ctx, "starting SQL liveness instance")
-	l.stopper.RunWorker(ctx, l.heartbeatLoop)
+	_ = l.stopper.RunAsyncTask(ctx, "slinstance", l.heartbeatLoop)
```
Comment (spaskob) [inline pkg/sql/sqlliveness/slinstance/slinstance.go:253]: for my education what is the difference?

**Comment by ajwerner** [comment]
@spaskob can I have another look at this?

**Review by ajwerner — COMMENTED** [review]
I just realized I didn't submit my responses :facepalm: 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52710#-:-MEwqInZ1Da9xHLy5YZ9:b-ghmola)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @spaskob)

___
*[pkg/sql/sqlliveness/sqlliveness.go, line 88 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52710#-MEYjFOTkAxny67Qqyb_-r1-88:-MEhcZZJ33JXdj2No45n:bbcnits) ([raw file](https://github.com/cockroachdb/cockroach/blob/317ae68cf1d068d2009c6d86c7f184a657767386/pkg/sql/sqlliveness/sqlliveness.go#L88)):*
<details><summary><i>Previously, spaskob (Spas Bojanov) wrote…</i></summary><blockquote>

why empty options?
</blockquote></details>

it uses the defaults which are sane. I'll comment. 
___
*[pkg/sql/sqlliveness/slinstance/slinstance.go, line 253 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/52710#-MEYjFOTkAxny67QqybY-r2-253:-MEhcmTO3iyigTZpeouY:b-z6xilj) ([raw file](https://github.com/cockroachdb/cockroach/blob/65ced8a0be98bb7b97987049c081785e2f6022c1/pkg/sql/sqlliveness/slinstance/slinstance.go#L253)):*
<details><summary><i>Previously, spaskob (Spas Bojanov) wrote…</i></summary><blockquote>

for my education what is the difference?
</blockquote></details>

The stopper API is too complex. The basic idea is that there are tasks and there are workers. Tasks launch workers. Quiescing means shut down all tasks. Stop means wait for all workers. Long-lived goroutines should be tasks and should respond to quiescing.

**Review by spaskob — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52710#-:-MEx3_Oz8sBNzMRy42bM:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @spaskob)

**Comment by ajwerner** [comment]
bors r=spaskob

**Pull request merged**